### PR TITLE
feat: added theming to List, ListItem, and ListIcon

### PR DESCRIPTION
**Pull request by jmiazga**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes
      /start features)

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

`List`, `ListItem`, and `ListIcon` are currently not themable.

Fixes: #2630

## What is the new behavior?

Now able to style `List`, `ListItem`, and `ListIcon` in the theme.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/theme/src/components/list.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const parts = ["container", "item", "icon"]
+
+const baseStyleContainer = {}
```
Comment (jmiazga) [inline packages/theme/src/components/list.ts:3]: Leaving these empty seams weird, but it does give us auto completion for the theme.

File: packages/layout/src/list.tsx [hunk 2]
```diff
@@ -97,14 +109,19 @@ if (__DEV__) {
...
  * ListItem
  *
  * Used to render a list item
  */
-export const ListItem = chakra("li")
+export const ListItem: React.FC<ListItemProps> = props => {
+  const styles = useStyles()
+  const LI = chakra("li")
+
+  return <LI {...props} __css={styles.item} />
```
Comment (TimKolberger) [inline packages/layout/src/list.tsx:123]: We could simplify that to `<chakra.li {...props} __css={styles.item} />`

File: packages/layout/src/list.tsx [hunk 3]
```diff
@@ -127,6 +146,7 @@ export const ListIcon = forwardRef<IconProps, "svg">(function ListIcon(
       display="inline"
       verticalAlign="text-bottom"
```
Comment (TimKolberger) [inline packages/layout/src/list.tsx:147]: Moving the default styles into the theme object would improve autocompletion when extending the theme and we would follow more the separation of concerns. The components just apply styles and manage their state and the styling is maintained in the theme.

File: packages/layout/src/list.tsx [hunk 4]
```diff
@@ -27,41 +31,49 @@ interface ListOptions {
...
 /**
  * List is used to display list items, it renders a `<ul>` by default.
  *
  * @see Docs https://chakra-ui.com/components/list
  */
 export const List = forwardRef<ListProps, "ul">(function List(props, ref) {
+  const styles = useMultiStyleConfig("List", props)
   const {
     children,
     styleType = "none",
```
Comment (TimKolberger) [inline packages/layout/src/list.tsx:48]: Please see "default styles" comment

**Review by TimKolberger — COMMENTED** [review]
Love seeing that more and more components are going to be themeable 💖

File: packages/layout/src/list.tsx (hunk 3 above)
Comment (jmiazga) [reply packages/layout/src/list.tsx:147]: Yeah we are all over the place with this. I wanted to chat with Sage about this at some point

File: packages/layout/src/list.tsx (hunk 4 above)
Comment (jmiazga) [reply packages/layout/src/list.tsx:48]: `styleType` is a prop, I wasn't able to get it to work with the theme as a default value. Let me know if you have any ideas.

Comment (TimKolberger) [reply packages/layout/src/list.tsx:48]: Oh, it is an alias for `listStyleType` - missed that 😓

**Pull request merged**